nucypher/blockchain/eth/events.py

```diff
@@ -100,13 +97,15 @@ def __init__(self,
             raise ValueError(f"Contract agent must be provided")
         if not event_name:
             raise ValueError(f"Event name must be provided")
+        if not from_block:
+            raise ValueError(f"From block must be provided")
 
         self.event_filter = agent.events[event_name]
         self.from_block = from_block
         self.to_block = to_block if to_block else agent.blockchain.client.block_number
```

Comment:
Can `to_block` be 0?

Reply:
Then perhaps it's better to check if `to_block is None` explicitly, because 0 would evaluate to `False` too.

Reply:
✔️